distribution_metadata.c

```diff
@@ -389,7 +391,7 @@ IsDistributedTable(Oid tableId)
 	 * The query below hits the partition metadata table, so if we don't detect
 	 * that and short-circuit, we'll get infinite recursion in the planner.
 	 */
-	if (tableId == partitionMetadataTableOid)
+	if (IsSystemNamespace(tableNamespaceOid) || tableId == partitionMetadataTableOid)
```

Comment:
Note to @sumedhpathak: the view-based `SELECT` gets rewritten with a subselect that hits the CitusDB metadata directly. So the recursion issue re-emerges. I know in #110 you liked me testing against the table `Oid` rather than the namespace of the target table, but I think it's clear that no catalog table ("system namespace") is ever distributed.

So adding this small check here fixes the CitusDB case.